### PR TITLE
🌐 Add French translation for `docs/fr/docs/history-design-future.md`

### PR DESCRIPTION
**Pull request by rjNemo**
Hello :wave,

add 🇫🇷 French translation for docs/- history-design-future.md (issue #1972).

Thank you!

### CODE REVIEW

File: docs/fr/mkdocs.yml [hunk 1]
```diff
@@ -54,6 +54,7 @@ nav:
 - features.md
 - fastapi-people.md
 - python-types.md
+- history-design-future.md
```
Comment (Smlep) [inline docs/fr/mkdocs.yml:57]: In the English doc, `history-design-future` comes between `alternatives.md` and `external-links.md`.

File: docs/fr/docs/history-design-future.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+# Histoire, conception et avenir
+
+Il y a quelque temps, <a href="https://github.com/tiangolo/fastapi/issues/3#issuecomment-454956920" class="external-link" target="_blank">un utilisateur de **FastAPI** a demandé</a> :
+
+> Quelle est l'histoire de ce projet ? Il semble être sorti de nulle part et est devenu génial en quelques semaines [...].
+
+Voici un peu bout de cette histoire.
```
Comment (Smlep) [inline docs/fr/docs/history-design-future.md:7]:
```suggestion
Voici un petit bout de cette histoire.
```

File: docs/fr/docs/history-design-future.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+<blockquote markdown="1">
+
+**FastAPI** n'existerait pas sans le travail antérieur d'autres personnes.
+
+Il y a eu de nombreux outils créés auparavant qui ont contribué à inspirer sa création.
+
+J'ai évité la création d'un nouveau framework pendant plusieurs années. J'ai d'abord essayé de résoudre toutes les fonctionnalités couvertes par **FastAPI** en utilisant de nombreux frameworks, plug-ins et outils différents.
+
+Mais à un moment donné, il n'y avait pas d'autre option que de créer quelque chose qui offre toutes ces fonctionnalités, en prenant les meilleures idées des outils précédents, et en les combinant de la meilleure façon possible, en utilisant des fonctionnalités du langage qui n'étaient même pas disponibles auparavant (indices de type pour Python 3.6+).
```
Comment (Smlep) [inline docs/fr/docs/history-design-future.md:27]: The french name for Python's `type hints` is `annotations de type`.

See the Python's french doc.

File: docs/fr/docs/history-design-future.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+</blockquote>
+
+## Recherche
+
+En utilisant toutes les alternatives précédentes, j'ai eu la chance d'apprendre de toutes, de prendre des idées, et de les combiner de la meilleure façon que j'ai pu trouver pour moi-même et les équipes de développeurs avec lesquelles j'ai travaillé.
+
+Par exemple, il était clair que l'idéal était de se baser sur les indices de type Python standard.
```
Comment (Smlep) [inline docs/fr/docs/history-design-future.md:35]:
```suggestion
Par exemple, il était clair que l'idéal était de se baser sur les annotations de type Python standard.
```

See the explanation from the previous comment.

File: docs/fr/docs/history-design-future.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+## Développement
+
+Au moment où j'ai commencé à créer **FastAPI** lui-même, la plupart des pièces étaient déjà en place, la conception était définie, les exigences et les outils étaient prêts, et la connaissance des normes et des spécifications était claire et fraîche.
+
+## Futur
+
+À ce stade, il est déjà clair que **FastAPI** et ses idées sont utiles pour de nombreuses personnes.
+
+Elle a été préférée aux autres solutions précédentes parce qu'elle convient mieux à de nombreux cas d'utilisation.
```
Comment (Smlep) [inline docs/fr/docs/history-design-future.md:71]:
```suggestion
Elle a été préférée aux solutions précédentes parce qu'elle convient mieux à de nombreux cas d'utilisation.
```
or 
```suggestion
Elle a été préférée aux autres solutions parce qu'elle convient mieux à de nombreux cas d'utilisation.
```

**Review by Smlep — CHANGES_REQUESTED** [review]

**Review by Smlep — APPROVED** [review]

File: docs/fr/docs/history-design-future.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+Il y a quelque temps, <a href="https://github.com/tiangolo/fastapi/issues/3#issuecomment-454956920" class="external-link" target="_blank">un utilisateur de **FastAPI** a demandé</a> :
+
+> Quelle est l'histoire de ce projet ? Il semble être sorti de nulle part et est devenu génial en quelques semaines [...].
+
+Voici un petit bout de cette histoire.
+
+## Alternatives
+
+Je crée des API avec des exigences complexes depuis plusieurs années (Machine Learning, systèmes distribués, jobs asynchrones, bases de données NoSQL, etc), en dirigeant plusieurs équipes de développeurs.
```
Comment (arthurio) [inline docs/fr/docs/history-design-future.md:11]: Do we use the plural form in french for API? ¯\_(ツ)_/¯ 

```suggestion
Je crée des APIs avec des exigences complexes depuis plusieurs années (Machine Learning, systèmes distribués, jobs asynchrones, bases de données NoSQL, etc), en dirigeant plusieurs équipes de développeurs.
```

File: docs/fr/docs/history-design-future.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+## Recherche
+
+En utilisant toutes les alternatives précédentes, j'ai eu la chance d'apprendre de toutes, de prendre des idées, et de les combiner de la meilleure façon que j'ai pu trouver pour moi-même et les équipes de développeurs avec lesquelles j'ai travaillé.
+
+Par exemple, il était clair que l'idéal était de se baser sur les annotations de type Python standard.
+
+De plus, la meilleure approche était d'utiliser des normes déjà existantes.
+
+Ainsi, avant même de commencer à coder **FastAPI**, j'ai passé plusieurs mois à étudier les spécifications d'OpenAPI, JSON Schema, OAuth2, etc. Comprendre leurs relations, leurs chevauchements et leurs différences.
```
Comment (arthurio) [inline docs/fr/docs/history-design-future.md:39]: chevauchement is correct but sounds a bit weird, WDYT?

```suggestion
Ainsi, avant même de commencer à coder **FastAPI**, j'ai passé plusieurs mois à étudier les spécifications d'OpenAPI, JSON Schema, OAuth2, etc. Comprendre leurs relations, leurs similarités et leurs différences.
```

File: docs/fr/docs/history-design-future.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+## Conception
+
+Ensuite, j'ai passé du temps à concevoir l'"API" de développeur que je voulais avoir en tant qu'utilisateur (en tant que développeur utilisant FastAPI).
+
+J'ai testé plusieurs idées dans les éditeurs Python les plus populaires : PyCharm, VS Code, les éditeurs basés sur Jedi.
+
+D'après le dernier <a href="https://www.jetbrains.com/research/python-developers-survey-2018/#development-tools" class="external-link" target="_blank">Python Developer Survey</a>, cela couvre environ 80% des utilisateurs.
```
Comment (arthurio) [inline docs/fr/docs/history-design-future.md:47]: Should we translate the Survey title?

```suggestion
D'après la dernière <a href="https://www.jetbrains.com/research/python-developers-survey-2018/#development-tools" class="external-link" target="_blank">Enquête Développeurs Python</a>, cela couvre environ 80% des utilisateurs.
```

**Review by arthurio — APPROVED** [review]
Très bonne traduction! Just couple of suggestions/questions.

File: docs/fr/docs/history-design-future.md (hunk 6 above)
Comment (rjNemo) [reply docs/fr/docs/history-design-future.md:11]: I don't think so.

**Comment by tiangolo** [comment]
Awesome, thanks @rjNemo! :cake: 

And thanks for the reviews @arthurio and @Smlep! :coffee:

**Pull request merged**